spec/requests/app_spec.rb

```diff
@@ -46,6 +48,41 @@
     end
   end
 
+  describe 'POST /user/residents' do
+    context 'when the file is valid' do
+      before do
+        allow(ResidentList::PDF).to receive(:new).and_return(
```

Comment:
Request specs are more of an "integration" spec, so we should try to use the real objects anytime we can. Since we do have a good PDF file and it's used here in the tests already, let's test against that instead of stubbing out `#load_residents`. I would remove this whole before block and test using a real instance of the ResidentList::PDF class.